### DISCUSSION TITLE
Why do onClick listeners not work with Static Site Generation?

### DISCUSSION THREAD

**Discussion by samwatts98** [vercel/next.js#16890] [Help] [answered]
Hi!
Forgive me if this is a silly question, still learning Next!

Take this code snippet for example:
```
import React from 'react'

export default function Home() {
  const [count, setCount] = React.useState(0);
  return (<button onClick={() => setCount(c => ++c)}>{count}</button>)
}
```
When I run the project in dev mode, this works as expected, on each click the value is incremented.
Now if I run **next build** and **next export** then the counter no longer works.

Essentially, I noticed this when porting a regular React project into a Next project, and the Dark Mode toggle button stopped working, along with the nav menu expand/collapse. 
Is this more than SSG is cut out for, and should just stick to a regular serverless deployment?

Thanks!

**Comment by mattcarlotta** [comment, marked as answer]
Unable to replicate. My guess is that your production server isn't serving JS assets. 

Working (local staging) example: https://github.com/mattcarlotta/with-static-counter

If you're still running into issues, please provide a working, minimal, and reproducible example of the problem.

**Reply by samwatts98** [reply]
Thanks for your quick response! 

Yes if I serve the static build via express it works as expected. 
However I had the intention of supplying the SSG build to GitHub pages, I was under the impression that the output of **next export** would work as it is without a production server. Perhaps I'm being too naive to expect that opening index.html from the build directory would work?

**Reply by mattcarlotta** [reply]
Github pages requires quite a few tweaks. You can take a look at the [official example](https://github.com/vercel/next.js/tree/canary/examples/gh-pages), but even that has some [asset serving issues](https://i.imgur.com/kgRVKgb.png). When I was looking into deploying on Github, it required a few additional tweaks not mentioned in the official example (can't remember exactly what), and ultimately I found it to be not worthwhile. Instead, I'd highly recommend deploying to vercel. It's free, it has continuous integration (pushes to the master automatically rebuild the website), custom build commands, and it allows custom domains (provided that you've purchased one). Overall, just a much better DX than Github pages.

**Reply by samwatts98** [reply]
Brilliant, thank you for the recommendation Matt! In hindsight now that I've migrated the project to Next it does make more sense to make the most of it and move to Vercel. 
Thanks again for your time!

**Comment by onmyway133** [comment]
@mattcarlotta I also have this problem. it works fine when in development, but once I export it, the onClick event does not trigger, and also the `react-tweet-embed` does not show

**Reply by Garamani** [reply]
@onmyway133 
I have the same problem, did you find the source of this problem?

**Reply by Garamani** [reply]
I found the problem: Having all CSS styles for all pages in a single file!

Creating a separate CSS file for each page solved the issue.

**Reply by thamaraiselvam** [reply]
What? how come CSS conflicts with JS?
